MushroomObserver/mushroom-observer
# Project date range

Adds Project start and end dates.
- Project admin sets start and end dates when creating or editing a Project.
- Those dates display on the Project page.
- If a Project is accepting new observations from members, then when a Project member creates an Observation:
  - If the ***current date*** is outside the Project date range, the Project is not automatically selected for the Observation;
  - If the User manually selects the Project and the ***Observation  date*** (`when`) is outside the Project date range, then
    - The User receives a warning (after clicking the Obs Create button) and a chance to correct by:
      - Deselecting the offending Project,
      - Changing the Obs date to within the Project Range
      - Overriding the warning by checking "Ignore Project Dates".
See  [Open Project Stories for MO](https://docs.google.com/document/d/1g8QMR6NzUvM4pjuAuBiI9QGZU_XNail-ocMu120UBxw/edit), Story 1, including comments

Delivers [Pivotal #186018784](https://www.pivotaltracker.com/story/show/186018784)

### Suggested manual test
1. Create a Project whose end (and start) dates are in the past and that’s accepting new observations from members.
2. Click Create Observation
3. Check the box for the project you just created.
4. Click the Create button
Expected Result: Warning including instructions on how to proceed.
5. Click the Create button
Expected Result: Ditto. (You didn’t fix the problem.).
6. Uncheck the project.
7. Click the Create button again.
Expected Result: Observation created.

8. Click Create Observation
9. Check the box for the project.
10. Click the Create button
Expected Result: Warning.
11. Check Ignore Project Dates
12. Click the Create button again.
Expected Result: Observation created && Observation added to project.

Same drill, this time fixing the problem by changing When to be in-range for the Project.
Expected Result: Observation created && Observation added to project.
